rust/src/delta_arrow.rs

```diff
@@ -95,7 +114,7 @@ impl TryFrom<&schema::SchemaDataType> for ArrowDataType {
                     }
                     "timestamp" => {
                         // Issue: https://github.com/delta-io/delta/issues/643
-                        Ok(ArrowDataType::Timestamp(TimeUnit::Nanosecond, None))
+                        Ok(ArrowDataType::Timestamp(TimeUnit::Microsecond, None))
```

Comment:
@fvaleye FYI, it looks like this PR reverted your nanosecond fix, which means your latest released python binding might be broken for tables created with default spark timestamp config :(

also cc @dispanser since he raised a similar issue.

Reply:
Let's move our discussion to https://github.com/delta-io/delta-rs/issues/441 to figure out a better fix on our end. I have a feeling that this needs to be handled separately within pyarrow and datafusion.